ui/src/Modals/RematchMedia/Index.jsx

```diff
@@ -0,0 +1,141 @@
+import { cloneElement, useCallback, useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import Modal from "react-modal";
+import { useDispatch, useSelector } from "react-redux";
+
+import { fetchMediaInfo } from "../../actions/media";
+import { NOTIFICATIONS_ADD } from "../../actions/types";
+import MediaTypeSelection from "./MediaTypeSelection";
+import DirSelection from "./DirSelection";
+import Field from "../../Pages/Auth/Field";
+import Button from "../../Components/Misc/Button";
+
+import Search from "./Search";
+import { RematchContext } from "./Context";
+
+import "./Index.scss";
+
+Modal.setAppElement("body");
+
+function RematchMediaModal(props) {
+  const dispatch = useDispatch();
+
+  const { id } = useParams();
+
+  const { token } = useSelector(store => ({
+    token: store.auth.token
+  }));
+
+  const [visible, setVisible] = useState(false);
+
+  const [query, setQuery] = useState("");
+  const [tmdbResults, setTmdbResults] = useState([]);
+  const [mediaType, setMediaType] = useState("movie");
+  const [tmdbID, setTmdbID] = useState();
+  const [error, setError] = useState("");
+  const [matching, setMatching] = useState(false);
+
+  const clearData = useCallback(() => {
+    setTmdbID();
+    setTmdbResults([]);
+    setError();
+    setQuery();
+  }, []);
+
+  // prevent scrolling behind Modal
+  useEffect(() => {
+    visible
+      ? document.body.style.overflow = "hidden"
+      : document.body.style.overflow = "unset";
+  }, [visible]);
+
+  const close = useCallback(() => {
+    setVisible(false);
+
+    if (props.cleanUp) {
+      props.cleanUp();
+    }
+  }, [props]);
+
+  const open = useCallback(() => {
+    setVisible(true);
+  }, []);
+
+  const rematch = useCallback(async () => {
+    if (!tmdbID || !mediaType) return;
+
+    setMatching(true);
+
+    const config = {
+      method: "PATCH",
+      headers: {
+        "authorization": token
+      }
+    };
+
+    const req = await fetch(`/api/v1/media/${id}/match?external_id=${tmdbID}&media_type=${mediaType}`, config);
+
+    if (req.status !== 200) {
+      setError(req.statusText);
+      return;
+    }
+
+    dispatch({
+      type: NOTIFICATIONS_ADD,
+      payload: {
+        msg: `Sucessfuly matched ${id}.`
+      }
+    });
+
+    clearData();
+    close();
+    dispatch(fetchMediaInfo(id));
```

Comment:
I think we might have to also fetch seasons here if the target for rematching is a show.